src/main/java/com/wavefront/opentracing/WavefrontSpan.java

```diff
@@ -63,7 +74,14 @@
         tracer.getWfInternalReporter().newCounter(
             new MetricName("spans.discarded", Collections.emptyMap()));
 
-    this.tags = (tags == null || tags.isEmpty()) ? null : new ArrayList<>();
+    this.tags = (globalTags == null || globalTags.isEmpty()) && (tags == null || tags.isEmpty()) ?
+      null : new ArrayList<>();
+    this.singleValuedTags = new HashMap<>();
```

Comment:
Can we avoid instantiating this for every span? Create only if needed within the `setTagObject` method.